storagemarket/lp2pimpl/net.go

```diff
@@ -325,12 +325,30 @@ func (p *DealProvider) getDealStatus(req types.DealStatusRequest) types.DealStat
 		return errResp("getting sector status from sealer")
 	}
 
+	sealingStatus := string(si.State)
+	var found bool
+
+	if storagemarket.IsFinalSealingState(si.State) {
+		if len(si.Deals) > 0 {
```

Comment:
This looks similar to the code in deal_execution that checks if the list of deals contains a particular deal id.
Maybe we can refactor this code into a function that can be called from all three places like
```storagemarket.HasDeal(si.Deals, pds.ChangeDealID)```